File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.py
import itertools
def solution(word):
    alphabet = ['A', 'E', 'I', 'O', 'U']
    answer = []
    for i in range(1, 6):
        for j in itertools.product(alphabet, repeat=i):
            answer.append(''.join(j))
    answer.sort()
    return answer.index(word)+1

    
    

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    const alphabet = ['','A', 'E', 'I', 'O', 'U']
    const len = alphabet.length
    const dict = []
    
    for(let a = 0; a < len; a++) {
        for(let b = 0; b < len; b++) {
            for(let c = 0; c < len; c++) {
                for(let d = 0; d < len; d++) {
                    for(let e = 0; e < len; e++) {
                        const makeWord = alphabet[a]+alphabet[b]+alphabet[c]+alphabet[d]+alphabet[e]
                        if(makeWord === '') continue
                        dict.push(makeWord)
                    }
                }
            }
        }
    }
    
    const uniqueDict = [...new Set(dict)]
    uniqueDict.sort()
    
    return uniqueDict.findIndex((element) => element === word)+1
}

